src/IndexRegistry.sol

```diff
@@ -244,65 +302,48 @@ contract IndexRegistry is IndexRegistryStorage {
+    /// @notice Returns the most recent _operatorCountHistory entry for the specified quorumNumber
+    /// @dev Reverts if the quorum does not exist
+    function getLatestQuorumUpdate(uint8 quorumNumber) external view returns (QuorumUpdate memory) {
+        return _latestQuorumUpdate(quorumNumber);
+    }
 
-        // blocknumber must be at or after the "index'th" entry's fromBlockNumber
-        require(
-            blockNumber >= quorumUpdate.fromBlockNumber, 
-            "IndexRegistry.getTotalOperatorsForQuorumAtBlockNumberByIndex: provided index is too far in the past for provided block number"
-        );
-        
-        // if there is an index update after the "index'th" update, the blocknumber must be before the next entry's fromBlockNumber
-        if (index != _totalOperatorsHistory[quorumNumber].length - 1){
-            QuorumUpdate memory nextQuorumUpdate = _totalOperatorsHistory[quorumNumber][index + 1];
-            require(
-                blockNumber < nextQuorumUpdate.fromBlockNumber, 
-                "IndexRegistry.getTotalOperatorsForQuorumAtBlockNumberByIndex: provided index is too far in the future for provided block number"
-            );
-        }
-        return quorumUpdate.numOperators;
+    /// @notice Returns the most recent _operatorCountHistory entry for the specified quorumNumber
+    /// @dev Reverts if there is no update for the given index
+    function getLatestOperatorUpdate(uint8 quorumNumber, uint32 index) external view returns (OperatorUpdate memory) {
+        return _latestIndexUpdate(quorumNumber, index);
     }
 
     /// @notice Returns an ordered list of operators of the services for the given `quorumNumber` at the given `blockNumber`
-    function getOperatorListForQuorumAtBlockNumber(
+    function getOperatorListAtBlockNumber(
         uint8 quorumNumber, 
         uint32 blockNumber
     ) external view returns (bytes32[] memory){
-        bytes32[] memory quorumOperatorList = new bytes32[](_getTotalOperatorsForQuorumAtBlockNumber(quorumNumber, blockNumber));
-        for (uint256 i = 0; i < quorumOperatorList.length; i++) {
-            quorumOperatorList[i] = _getOperatorIdAtIndexForQuorumAtBlockNumber(uint32(i), quorumNumber, blockNumber);
+        uint32 operatorCount = _operatorCountAtBlockNumber(quorumNumber, blockNumber);
+        bytes32[] memory operatorList = new bytes32[](operatorCount);
+        for (uint256 i = 0; i < operatorCount; i++) {
+            operatorList[i] = _operatorIdForIndexAtBlockNumber(uint32(i), quorumNumber, blockNumber);
             require(
-                quorumOperatorList[i] != OPERATOR_DOES_NOT_EXIST_ID, 
-                "IndexRegistry.getOperatorListForQuorumAtBlockNumber: operator does not exist at the given block number"
+                operatorList[i] != OPERATOR_DOES_NOT_EXIST_ID, 
+                "IndexRegistry.getOperatorListAtBlockNumber: operator does not exist at the given block number"
```

Comment:
this is effectively an assert here, right?
seems like by design there should be no path that ever allows hitting this require statement. If it's triggered, then at least one system property has been broken?

Reply:
1. The lookup returns this value if the index was never used at the blocknumber.
2. When we remove an operator, the final entry in the index history ends up being this value